src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -83,10 +83,7 @@ public function import($resource, $type = null, $ignoreErrors = false, $sourceRe
             $loader = $this->resolve($resource, $type);
 
             if ($loader instanceof self && null !== $this->currentDir) {
-                // we fallback to the current locator to keep BC
-                // as some some loaders do not call the parent __construct()
-                // @deprecated should be removed in 3.0
-                $locator = $loader->getLocator() ?: $this->locator;
+                $locator = $loader->getLocator();
                 $resource = $locator->locate($resource, $this->currentDir, false);
```

Comment:
Doesn't that introduce a possible failure? As apparently not all `$loader->getLocator()` returned something (otherwise the fallback would have never been called), this may now fail as `$locator` can be empty?
Or was this changed in `$loader->getLocator()`, so that it always returns a locator?